ws-client/src/client.rs

```diff
@@ -115,15 +150,33 @@ impl WsClient {
 		let max_concurrent_requests = config.max_concurrent_requests;
 		let request_timeout = config.request_timeout;
 		let (to_back, from_front) = mpsc::channel(config.max_concurrent_requests);
+		let (err_tx, err_rx) = oneshot::channel();
 
 		let (sender, receiver) = jsonrpc_transport::websocket_connection(config.clone())
 			.await
 			.map_err(|e| Error::TransportError(Box::new(e)))?;
 
 		async_std::task::spawn(async move {
-			background_task(sender, receiver, from_front, max_capacity_per_subscription, max_concurrent_requests).await;
+			background_task(
+				sender,
```

Comment:
maybe introduce a new type for all this params ^^

Reply:
Maybe, but it's not too bad yet. Let's hold off on that for now?

Reply:
yeah, let's ignore for now :)